src/mlpack/tests/split_data_test.cpp

```diff
@@ -121,6 +121,30 @@ TEST_CASE("SplitDataResultMat", "[SplitDataTest]")
   CheckMatrices(input, concat);
 }
 
+TEST_CASE("SplitDataResultField", "[SplitDataTest]")
+{
+  field<mat> input(1, 2);
+
+  mat matA(2, 10);
+  mat matB(2, 10);
+
+  size_t count = 0; // Counter for unique sequential values.
+  matA.imbue([&count]() { return ++count; });
+  matB.imbue([&count]() { return ++count; });
+
+  input(0, 0) = matA;
+  input(0, 1) = matB;
+
+  const auto value = Split(input, 0.5, false);
+  REQUIRE(std::get<0>(value).n_cols == 1); // Train data.
+  REQUIRE(std::get<1>(value).n_cols == 1); // Test data.
+
+  field<mat> concat = {std::get<0>(value)(0), std::get<1>(value)(0)};
+  // Order matters here.
+  CheckFields(input, concat);
+}
+
+
```

Comment:
```suggestion
```

No need for an extra empty line. :)

Also, do you think you could add a test case for when the labels are not a `Row` but instead a matrix or something like that?  (i.e. let's make sure we test all the new functionality you're adding.)